front/lib/api/assistant/agent.ts

```diff
@@ -718,29 +732,37 @@ export async function* runMultiActionsAgent(
   const chainOfThought = tokenEmitter.getChainOfThought();
   const content = tokenEmitter.getContent();
 
-  if (chainOfThought?.length || content?.length) {
+  // TODO(@fontanierh): migrate to content.
+  if (chainOfThought?.length) {
     yield {
       type: "agent_chain_of_thought",
       created: Date.now(),
       configurationId: agentConfiguration.sId,
       messageId: agentMessage.sId,
       message: agentMessage,
+      chainOfThought,
+    } satisfies AgentChainOfThoughtEvent;
+  }
 
-      // All content here was generated before a tool use and is not proper generation content
-      // and can therefore be safely assumed to be reflection from the model before using a tool.
-      // In practice, we should never have both chainOfThought and content.
-      // It is not completely impossible that eg Anthropic decides to emit part of the
-      // CoT between `<thinking>` XML tags and the rest outside of any tag.
-      chainOfThought: removeNulls([chainOfThought, content]).join("\n"),
-    };
+  // This content was emitted despite the tools use.
+  // We emit is as a separate event to keep track of it in
+  // the AgentMessageContent table.
+  if (content?.length) {
+    yield {
+      type: "agent_message_content",
```

Comment:
this content is post tokenEmitter and as such not the raw content. Or this will trigger the creation of an AgentMessageContent. Not sure I understand as this does not seem aligned with what we said earlier?